### DISCUSSION TITLE
Working with Vierstra motif clusters

### DISCUSSION THREAD

**Discussion by jindalk** [GreenleafLab/ArchR#1364] [Feature Requests]
I've been using the vierstra motif PWMs for chromVAR analysis. Is there a way of calculating chromVAR scores on a per motif cluster basis (by somehow merging PWMs of motifs from the same cluster)? Or can I combine chromVAR scores from motifs of the same cluster, after computing Deviations?

One possible solution I could think of is to download the archetype bed file from here https://resources.altius.org/~jvierstra/projects/motif-clustering/releases/v1.0/ , split it by motif cluster and add them as a peak annotation.

Thanks

**Comment by rcorces** [comment]
I'm not sure I fully understand your question. Can you read this issue post and let me know if this answers it?
https://github.com/GreenleafLab/ArchR/issues/214

**Reply by jindalk** [reply]
Hi Ryan, sorry for not being more clear. Yes, I read that post earlier and have been using the Vierstra PWM Rds that was posted there. From what I understand, the file contains 1 PWM for each of the ~2100 motifs and would allow me to compute deviations for each motif. However, I am trying to compute deviations with each motif cluster (and not motif) as a feature (so 286 total features). I hope that makes more sense.

Thanks

**Reply by rcorces** [reply]
I'm not sure how to accomplish what you are trying to accomplish. My interpretation of this is that it isnt super straightforward to capture a single Vierstra archetype (what you call a motif cluser) in a single PWM. I imagine if that were straightforward, then Jeff Vierstra would have done it and created a catalog of archetype PWMs. My understanding is that the way he did the analysis, he scanned for motif occurrences for each motif individually and then de-duplicated based on the archetypes.

I think the first question you should ask is how different are the chromVar scores across the different members of an archetype/motif cluster. Perhaps taking the mean or median would suffice. My intuition is that the variation in chromVAR deviations should be quite small across archetype/motif clusters. 

For reference, the cluster residence of the different motifs is here:
https://resources.altius.org/~jvierstra/projects/motif-clustering/releases/v1.0/motif_annotations.xlsx
and the PWMs that we provide have that information coded in the name

**Reply by jindalk** [reply]
Yeah that's fair, it did seem fairly complicated to me as well. I can certainly look into intra-archetype variance to see if simply aggregating chromVAR deviation z-scores within archetypes would suffice.

Thanks!

**Reply by rcorces** [reply]
@jindalk - In case you arent following Jeff Vierstra on twitter, he has made consensus PWMs in MEME format
https://resources.altius.org/~jvierstra/projects/motif-clustering-v2.0beta/
https://twitter.com/jeffvierstra/status/1514973966045048835?s=20&t=Grg3kA7ddZWrIH7tl1zBIQ

We will work on incorporating these consensus PWMs into ArchR.

**Reply by jindalk** [reply]
Hi Ryan, I saw that this morning. Very exciting! :D Looking forward to trying them out in ArchR

**Reply by rcorces** [reply]
This now is functional in `release_1.0.2`. I've changed how the "Vierstra" motif databases are accessed. You no longer need to download these separately. You can now get these using `motifSet="vierstra"` and `collection="archetype"` for the archetype clustered pwms or `collection = "individual"` for the previous individual pwms.

See new announcement here - https://github.com/GreenleafLab/ArchR/discussions/1386

**Reply by rcorces** [reply]
Please let me know if you test this out and it works in your hands. In `release_1.0.2` there are some parameter changes, including `annoName` instead of the old `name`.
```
#' @param ArchRProj An `ArchRProject` object.
#' @param motifSet The motif set to be used for annotation. Options include: (i) "JASPAR2016", "JASPAR2018", "JASPAR2020"
#' which gives the 2016, 2018 or 2020 version of JASPAR motifs, (ii) one of "cisbp", "encode", or "homer" which gives the
#' corresponding motif sets from the `chromVAR` package, or (iii) "vierstra" which gives the clustered archetype motifs
#' created by Jeff Vierstra (https://github.com/jvierstra/motif-clustering). 
#' @param annoName The name of the `peakAnnotation` object to be stored in the provided `ArchRProject`
#' @param species The name of the species relevant to the supplied `ArchRProject`. This is used for identifying which motif to be
#' used from CisBP/JASPAR. By default, this function will attempt to guess the species based on the value from `getGenome()`.
#' @param collection If one of the JASPAR motif sets is used via `motifSet`, this parameter allows you to indicate the JASPAR
#' collection to be used. See `getMatrixSet()` from `TFBSTools` for all options to supply for collection. If `motifSet` is
#' "vierstra", then this must either be "archetype" (for the v2 clustered models) or "individual" (for the original v1 individual motif models). 
#' NOTE: vierstra archetype motifs are currently in beta and have not been finalized by Jeff Vierstra.
#' @param motifPWMs A custom set of motif PWMs as a PWMList for adding motif annotations.
#' @param cutOff The p-value cutoff to be used for motif search. The p-value is determined vs a background set of sequences
#' (see `MOODS` for more details on this determination).
#' @param width The width in basepairs to consider for motif matches. See the `motimatchr` package for more information.
#' @param version An integer specifying version 1 or version 2 of chromVARmotifs see github for more info GreenleafLab/chromVARmotifs.
#' @param force A boolean value indicating whether to force the `peakAnnotation` object indicated by `annoName` to be overwritten if
#' it already exists in the given `ArchRProject`.
#' @param logFile The path to a file to be used for logging ArchR output.
#' @param ... Additional parameters to be passed to `TFBSTools::getMatrixSet` for getting a PWM object.
```

**Reply by jindalk** [reply]
wow, great work adding it to ArchR natively! Will report back when I try it

**Reply by jindalk** [reply]
I just tried it and it seems to be working really well! Thanks again for adding this to ArchR. One quick suggestion: Perhaps ArchR could print this URL when vierstra v2 annotation is added? https://resources.altius.org/~jvierstra/projects/motif-clustering-v2.0beta/

When working with archetype results, I would keep going back to this page to make sense of the results. This isn't critical to any functionality, just for added convenience.

**Reply by rcorces** [reply]
sure. I've added that:
```
> projHeme5 <- addMotifAnnotations(ArchRProj = projHeme5, motifSet = "Vierstra", collection = "archetype")
ArchR logging to : ArchRLogs/ArchR-addMotifAnnotations-9d612bb3443a-Date-2022-05-06_Time-11-54-16.log
If there is an issue, please report to github with logFile!
2022-05-06 11:54:16 : Gettting Motif Set, Species : , 0.003 mins elapsed.
Using Vierstra v2.1beta motifs. See https://resources.altius.org/~jvierstra/projects/motif-clustering-v2.1beta/ for more details.
Motif file Vierstra_Archetype_Motifs_v2.1.rds does not exist! Downloading..
trying URL 'https://jeffgranja.s3.amazonaws.com/ArchR/Annotations/Vierstra_Archetype_Motifs_v2.1.rds'
Content type 'binary/octet-stream' length 53319 bytes (52 KB)
==================================================
```

Worth noting that there is already a v2.1beta of the Vierstra motifs due to a bug that was found in their code for the v2.0beta and ArchR has been updated accordingly to use v2.1beta.

**Reply by jindalk** [reply]
Awesome, thank you!
